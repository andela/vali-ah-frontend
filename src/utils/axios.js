/* eslint-disable no-param-reassign */
/**
 * An axios proxy to attach the base url and authentication token
 */
import axios from 'axios';

const axiosInstance = axios.create({
  baseURL: process.env.API_ROUTE || 'http://localhost:3000',
});

axiosInstance.interceptors.request.use((config) => {
  config.headers.Authorization = `BEARER ${localStorage.getItem('token')}`;

  return config;
});

axiosInstance.interceptors.response.use((config) => {
  config.responseBody = config?.data?.data;

  return config;
});

export default axiosInstance;
