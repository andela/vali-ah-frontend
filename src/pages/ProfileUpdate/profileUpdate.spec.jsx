// react libraries
import React from 'react';
import axios from 'axios';

// pages
import { ProfileUpdate } from './index';

describe('Profile update', () => {
  const props = {
    profile: {
      loading: false,
      data: [],
      errors: null,
    },
    history: { push: jest.fn() },
    getProfileRequestUpdateSuccess: jest.fn(),
  };

  let wrapper;
  let label;

  beforeEach(async () => {
    const { data } = await axios.post(
      'https://vali-1kbideas-staging.herokuapp.com/api/v1/auth/signin',
      {
        password: 'lovem!588',
        email: 'myjoy@gmail.com',
      },
    );

    const { token } = data.data;

    localStorage.setItem('token', token);
    wrapper = mount(
      <ProfileUpdate {...props}>
        <div className="profile-container" />
      </ProfileUpdate>,
    );
    label = wrapper.find('label');
  });

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should ascertain that all the divs are existing', () => {
    expect(wrapper.find('profile-container_header')).toHaveLength(0);
    expect(wrapper.find('profile-container_header')).toHaveLength(0);
    expect(wrapper.find('profile-container_mini_header')).toHaveLength(0);
    expect(wrapper.find('profile-container_image-update')).toHaveLength(0);
    expect(wrapper.find('container_image_cover')).toHaveLength(0);
  });

  it('should submit a form with correct input', () => {
    window.URL.createObjectURL = jest.fn();
    const inputFirstName = wrapper.find('input').at(1);
    const inputLastname = wrapper.find('#lastname');
    const inputUsername = wrapper.find('input').at(3);
    const inputBio = wrapper.find('.update-bio');
    const form = wrapper.find('#updateform');
    const file = new File(['(⌐□_□)'], 'peter.png', { type: 'image/png' });
    const selectImage = wrapper.find('input').at(0);

    selectImage.simulate('change', {
      target: { files: [file], name: 'image' },
    });

    inputFirstName.simulate('change', {
      target: { value: 'Welliams', name: 'firstName' },
    });
    inputLastname.simulate('change', {
      target: { value: 'Lotsfor', name: 'lastName' },
    });
    inputUsername.simulate('change', {
      target: { value: 'Billillsz', name: 'userName' },
    });
    inputBio.simulate('change', {
      target: { value: 'in this life just think', name: 'bio' },
    });
    form.simulate('submit');
    const newState = wrapper.state();

    expect(label).toHaveLength(5);
    expect(newState.data.firstName).toEqual('Welliams');
    expect(newState.data.lastName).toEqual('Lotsfor');
    expect(newState.data.userName).toEqual('Billillsz');
    expect(newState.data.bio).toEqual('in this life just think');
    expect(newState.data.image).toBeInstanceOf(File);
  });
});
