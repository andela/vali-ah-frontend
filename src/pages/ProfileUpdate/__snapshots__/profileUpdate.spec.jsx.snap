// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile update should render properly 1`] = `
<ProfileUpdate
  getProfileRequestUpdateSuccess={[MockFunction]}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  profile={
    Object {
      "data": Array [],
      "errors": null,
      "loading": false,
    }
  }
>
  <div>
    <div
      className="profile-container"
    >
      <div
        className="profile-container__header"
      >
        <div
          className="profile-container__mini-header"
        />
        <div
          className="profile-container__image profile-container__image-update"
        >
          <div
            className="profile-container__image-cover"
          >
            <input
              accept=".png, .jpg, .jpeg"
              id="image-upload"
              onChange={[Function]}
              type="file"
            />
            <label
              htmlFor="image-upload"
            >
               
              <img
                alt=""
                className="profile-container__image-camera"
                src="test-file-stub"
              />
            </label>
          </div>
          <div
            className="profile-container__image-placeholder-update"
          >
            <img
              alt="Profile avatar of "
              className="profile-container__image-preview"
              id="profile-img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          className="profile-container__name"
        >
          <p
            className="username"
          >
              
          </p>
        </div>
        <div
          className="profile-container__bio-update"
        >
          <p
            id="bio"
          />
        </div>
      </div>
      <div
        className="profile-container__details"
      >
        <form
          className="ui form"
          id="updateform"
          onSubmit={[Function]}
        >
          <div
            className="field"
          >
            <label
              htmlFor="firstname"
            >
              First Name
              <input
                className="firstName"
                id="firstname"
                name="firstName"
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="First Name"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            className="field"
          >
            <label
              htmlFor="lastName"
            >
              Last Name
              <input
                className="lastname"
                id="lastname"
                name="lastName"
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Last Name"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            className="field"
          >
            <label
              htmlFor="username"
            >
              Username
              <input
                className="username"
                id="username"
                name="userName"
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Username"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            className="field"
          >
            <label
              htmlFor="bio"
            >
              Short Bio
              <textarea
                className="update-bio"
                id="bio"
                name="bio"
                onChange={[Function]}
                onFocus={[Function]}
                rows="2"
                value=""
              />
            </label>
          </div>
          <button
            className="ui right floated button orange button"
            disabled={false}
            type="submit"
          >
            Update
          </button>
        </form>
      </div>
    </div>
  </div>
</ProfileUpdate>
`;
